webrtc/pending-promise-after-close.html

```diff
@@ -0,0 +1,81 @@
+<!doctype html>
+<title>Test pending promises behavior when RTCPeerConnection object is closed</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script type="text/javascript">
+  'use strict';
+
+  function test_never_resolved(test_func, testName) {
+    async_test(t => {
+      const pc1 = new RTCPeerConnection();
+      const closePc1 = () => pc1.close();
+
+      test_func(pc1, closePc1)
+      .then(
+        t.step_func(result => {
+          assert_unreached(`Pending promise should never be resolved. Instead it is fulfilled with: ${result}`);
+        }),
+        t.step_func(err => {
+         assert_unreached(`Pending promise should never be resolved. Instead it is rejected with: ${err}`);
+        }));
+
+      t.step_timeout(() => {
+        t.done();
+      }, 1000);
+
+      const pc2 = new RTCPeerConnection();
+      const closePc2 = () => {}; // noop (don't close)
+
+      test_func(pc2, closePc2)
+      .then(
+        t.step_func_done(),
+        t.step_func(err => {
+          assert_unreached(`Operating on unclosed peer connection should succeed. Instead it is rejected with: ${err}`);
+        }));
```

Comment:
I have updated the code with some comments to clarify, and removed the timeout function waiting. If you find the test helper to be too complicated, I am happy to revert back to the old approach of using timeout and create one peer connection only.